### PR TITLE
Patching transactions

### PR DESCRIPTION
**Pull request by fenollp**

### CODE REVIEW

**Comment by fenollp** [comment]
@jamesaimonetti should be good now

File: core/kazoo_modb-1.0.0/src/kazoo_modb.erl [hunk 1]
```diff
@@ -319,3 +320,20 @@ delete_if_orphaned(AccountMODb, 'true') ->
...
     Succeeded.
+
+
+%% @public
+-spec get_range(ne_binary(), gregorian_seconds(), gregorian_seconds()) ->
+                       ne_binaries().
+get_range(AccountId, From, To) ->
+    {{FromYear, FromMonth, _}, _} = calendar:gregorian_seconds_to_datetime(From),
+    {{ToYear, ToMonth, _}, _} = calendar:gregorian_seconds_to_datetime(To),
+    Seq = crossbar_util:generate_year_month_sequence({FromYear, FromMonth}
```
Comment (jamesaimonetti) [inline core/kazoo_modb-1.0.0/src/kazoo_modb.erl]: Check if anything else in Crossbar uses this util function. If not, we should move it here too. Core libs should not be calling application modules!

File: core/whistle_transactions-1.0.0/src/wh_transactions.erl [hunk 2]
```diff
@@ -131,132 +142,173 @@ filter_by_reason(<<"only_calls">>, Transactions) ->
...
+              {Account, Year, Month} = kazoo_modb_util:split_account_mod(MODb),
+              [{'startkey', From}
+               ,{'endkey', To}
+               ,{'year', Year}
+               ,{'month', Month}
+               ,'include_docs'
+              ]
+          end || MODb <- kazoo_modb:get_range(Account, From, To)
+        ],
+    case length(ViewOptionsList) =< 3 of
```
Comment (jamesaimonetti) [inline core/whistle_transactions-1.0.0/src/wh_transactions.erl]: I wouldn't worry about this check; `cb_modules_util:range_view_options/2` enforces this limit. The core lib shouldn't be concerned with it.

**Comment by fenollp** [comment]
@jamesaimonetti goodagain

File: core/whistle-1.0.0/include/wh_types.hrl [hunk 3]
```diff
@@ -106,6 +106,8 @@
 -type wh_iso_week() :: calendar:yearweeknum(). %%{wh_year(), wh_weeknum()}.
 -type gregorian_seconds() :: pos_integer().
 -type unix_seconds() :: pos_integer().
+-type api_seconds() :: 'undefined' | gregorian_seconds().
+-type year_month_tuple() :: {gregorian_seconds(),gregorian_seconds()}.
```
Comment (jamesaimonetti) [inline core/whistle-1.0.0/include/wh_types.hrl]: Should be `{wh_year(), wh_month()}` I think

**Pull request merged**